torch/_guards.py

```diff
@@ -0,0 +1,142 @@
+import dataclasses
+import enum
+from typing import Callable, List, Optional
+
+
```

Comment:
I think this file now deserves a comment saying what it does, and why it lives here and not `torch/_dynamo`

Reply:
yes